packages/maskbook/src/plugins/Wallet/UI/WalletConnectQRCodeDialog/index.tsx

```diff
@@ -0,0 +1,86 @@
+import { useCallback, useEffect, useState } from 'react'
+import { Button, createStyles, DialogActions, DialogContent, makeStyles } from '@material-ui/core'
+import { useRemoteControlledDialog } from '../../../../utils/hooks/useRemoteControlledDialog'
+import { WalletMessages } from '../../messages'
+import Services from '../../../../extension/service'
+import { InjectedDialog } from '../../../../components/shared/InjectedDialog'
+import { SafariPlatform } from './SafariPlatform'
+import { FirefoxPlatform } from './FirefoxPlatform'
+import { QRCodeModel } from './QRCodeModel'
+import { useI18N } from '../../../../utils/i18n-next-ui'
+
+const useStyles = makeStyles(() =>
+    createStyles({
+        container: {
+            display: 'flex',
+            flexDirection: 'column',
+            alignItems: 'center',
+            justifyContent: 'center',
+        },
+    }),
+)
+
+export const WalletConnectQRCodeDialog: React.FC = () => {
```

Comment:
I suggest using function declaration in this case because it doesn't have TDZ

Reply:
TDZ?

Reply:
Temporal Dead Zone